app/controllers/api/v3/courses_controller.rb

```diff
@@ -3,9 +3,15 @@ module V3
     class CoursesController < API::V3::ApplicationController
       before_action :build_recruitment_cycle
       before_action :build_provider
-      before_action :build_course
+
+      def index
+        @courses = @provider.courses
+        render jsonapi: @courses, include: params[:include]
+      end
 
       def show
+        @course = @provider.courses.find_by!(course_code: params[:code].upcase)
+
         if @course.is_published?
           render jsonapi: @course, fields: fields_param
```

Comment:
Can you add a `params[:include]`? I think we / Apply need it to request locations

Reply:
Done.